utils/build-script

```diff
@@ -977,42 +975,63 @@ def main_preset():
 
     if len(args.preset_file_names) == 0:
         args.preset_file_names = [
-            os.path.join(HOME, ".swift-build-presets"),
             os.path.join(
                 SWIFT_SOURCE_ROOT, SWIFT_REPO_NAME, "utils",
                 "build-presets.ini")
         ]
 
+        user_presets_file = os.path.join(HOME, '.swift-build-presets')
+        if os.path.isfile(user_presets_file):
+            args.preset_file_names.append(user_presets_file)
```

Comment:
This is technically a behavior change.
If the both files has presets in a *same* name, `ConfigParser` merges that section. e.g.
```ini
# preset1.ini
[section]
opt1 = True
opt2 = foo
```
```ini
# preset2.ini
[section]
opt1 = False
opt3 = bar
```
`parser.read(['preset1.ini', 'preset2.ini'])` will be like:
```javascript
{
   "section": {
     "opt1": "False", // NOTE: preset2.ini overwrites this
     "opt2": "foo",
     "opt3": "bar"
   }
}
```

Reply:
I'm not convinced this is a behavior change. The old presets parsing code could read in more than one file at a time as well (namely the [`_load_preset_files_impl`](https://github.com/apple/swift/blob/c03533314d1732c6d89b559000fe788c84680134/utils/swift_build_support/swift_build_support/SwiftBuildSupport.py#L91) function.

Reply:
I meant the order in the `preset_file_names` might affect the result of the parsing.
In your implementation, the order is `['~/.swift-build-presets', 'utils/build-presets.ini']`, whereas the current master implementation is `['utils/build-presets.ini', '~/.swift-build-presets']`.
When `~/.swift-build-presets` declares an option with the same name in the same section name in `utils/build-presets.ini`, in the current implementation, it will be ignored because it will be overwritten by the latter file.

To keep the current behavior, the code should insert `~/.swift-build-presets` at the top of the list.
```python
        if os.path.isfile(user_presets_file):
             args.preset_file_names.insert(0, user_presets_file)
```

That's said, I think your implementation is OK because user defined (`~/.swift-build-presets`) value should be able to override the default value.

Reply:
I would rather keep this functionality change (however slight) since you make a good point. User presets should be able to override the built-in ones.